### PR TITLE
Add isSiteKitScreen utility function, ensure that the permissions modal is only visible on Site Kit screens.

### PR DESCRIPTION
**Pull request by makiost**
## Summary

Addresses issue:

- #5424 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/util/is-site-kit-screen.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * Internal dependencies
+ */
+import { SITE_KIT_VIEW_CONTEXTS } from '../googlesitekit/constants';
+
+/**
+ * Checks whether or not the current viewContext is a Site Kit screen.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} viewContext THe view-context.
```
Comment (jimmymadon) [inline assets/js/util/is-site-kit-screen.js:34]:
```suggestion
 * @param {string} viewContext The view-context.
```

File: assets/js/util/is-site-kit-screen.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Utility function to check whether or not a view-context is a site kit view.
```
Comment (jimmymadon) [inline assets/js/util/is-site-kit-screen.js:2]:
```suggestion
 * Utility function to check whether or not a view-context is a Site Kit view.
```

**Review by jimmymadon — CHANGES_REQUESTED** [review]
Thanks @makiost - this looks great! Just added some very minor doc suggestions and a question on using lodash.

File: assets/js/util/is-site-kit-screen.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Checks whether or not the current viewContext is a Site Kit screen.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} viewContext THe view-context.
+ * @return {boolean} TRUE if the passed view-context is a site kit view; otherwise FALSE.
+ */
+const isSiteKitScreen = ( viewContext ) =>
+	includes( SITE_KIT_VIEW_CONTEXTS, viewContext );
```
[the remark this answers is not in the document]
Comment (makiost) [reply assets/js/util/is-site-kit-screen.js]: My bad, totally forgot about the vanilla includes method! 🥴

**Review by jimmymadon — APPROVED** [review]
Thanks @makiost - LGTM!

File: assets/js/util/is-site-kit-screen.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Checks whether or not the current viewContext is a Site Kit screen.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} viewContext The view-context.
+ * @return {boolean} TRUE if the passed view-context is a site kit view; otherwise FALSE.
+ */
+const isSiteKitScreen = ( viewContext ) =>
```
Comment (eugene-manuilov) [inline assets/js/util/is-site-kit-screen.js:32]: Thanks, @makiost. In general, looks good to me, but could you please add js tests for this function?

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

File: assets/js/util/is-site-kit-screen.test.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+			true
+		);
+		expect( isSiteKitScreen( VIEW_CONTEXT_PAGE_DASHBOARD_VIEW_ONLY ) ).toBe(
+			true
+		);
+		expect( isSiteKitScreen( VIEW_CONTEXT_USER_INPUT ) ).toBe( true );
+		expect( isSiteKitScreen( VIEW_CONTEXT_DASHBOARD_SPLASH ) ).toBe( true );
+		expect( isSiteKitScreen( VIEW_CONTEXT_SETTINGS ) ).toBe( true );
+		expect( isSiteKitScreen( VIEW_CONTEXT_MODULE_SETUP ) ).toBe( true );
+	} );
```
Comment (eugene-manuilov) [inline assets/js/util/is-site-kit-screen.test.js:50]: We need to use `it.each( [] )()` approach to make each value a separate test:

```js
it.each( [
    [ 'VIEW_CONTEXT_DASHBOARD', VIEW_CONTEXT_DASHBOARD ],
    ...
] )( 'should return TRUE for %s screen', ( _, screen ) => {
    expect( isSiteKitScreen( screen ) ).toBe( true );
} );
```

Same applies to other tests as well.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks for adding tests, @makiost. Added one comments to ask you to restruct tests to use the `it.each()()` approach. Please, take a look.

File: assets/js/util/is-site-kit-screen.test.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		expect( isSiteKitScreen( null ) ).toBe( false );
+		expect( isSiteKitScreen( undefined ) ).toBe( false );
+		expect( isSiteKitScreen( 0 ) ).toBe( false );
+		expect( isSiteKitScreen( 1 ) ).toBe( false );
+		expect( isSiteKitScreen( true ) ).toBe( false );
+		expect( isSiteKitScreen( false ) ).toBe( false );
+		expect( isSiteKitScreen( {} ) ).toBe( false );
+		expect( isSiteKitScreen( [] ) ).toBe( false );
+		expect( isSiteKitScreen( function () {} ) ).toBe( false );
+	} );
```
Comment (eugene-manuilov) [inline assets/js/util/is-site-kit-screen.test.js:77]: @makiost we need to use `it.each()()` for this test too 😄 

```js
it.each( [
    [ 'a string', 'foo' ],
    [ 'an empty object', {} ],
    [ 'a number', 1 ],
    ...
] )( 'should return FALSE for %s', ( _, screen ) => {
    expect( isSiteKitScreen( screen ) ).toBe( false );
} );
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

**Review by eugene-manuilov — APPROVED** [review]
Nice! Thanks, @makiost.

**Pull request merged**